File: fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/method/Unsubscribe.java
package com.github.dedis.popstellar.model.network.method;

import com.github.dedis.popstellar.model.Immutable;
import com.github.dedis.popstellar.model.objects.Channel;

/** Unsubscribe from a channel */
@Immutable
public final class Unsubscribe extends Query {

  public Unsubscribe(Channel channel, int id) {
    super(channel, id);
  }

  @Override
  public String getMethod() {
    return Method.UNSUBSCRIBE.getMethod();
  }
}

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/method/Unsubscribe.kt
package com.github.dedis.popstellar.model.network.method

import com.github.dedis.popstellar.model.Immutable
import com.github.dedis.popstellar.model.objects.Channel

/** Unsubscribe from a channel */
@Immutable
class Unsubscribe(channel: Channel?, id: Int) : Query(channel, id) {
  override val method: String
    get() = Method.UNSUBSCRIBE.method
}
